pkg/management/component.go

```diff
@@ -152,19 +167,60 @@ func (c *component) CreateUser(ctx context.Context, realm string, user api.UserR
 		userRep.Attributes = &attributes
 	}
 
-	locationURL, err := c.keycloakClient.CreateUser(accessToken, realm, userRep)
+	locationURL, err := c.keycloakClient.CreateUser(accessToken, realmName, userRep)
 
 	if err != nil {
 		return "", err
 	}
 
+	//store the API call into the DB
+	var event = make(map[string]string)
+	event["ct_event_type"] = "API_ACCOUNT_CREATION"
```

Comment:
In component.go, we don't want to many business unrelated code in our functions. We should try to have the less "invasive" code as possible here. Could we create functions like :
```
var event map[string]string = events.CreateEventMap("API_ACCOUNT_CREATION")
events.AddEventValues("realm_name", realmName, "user_id", user.Id, "username", user.Username, "origin", "back-office", "audit-time", time.Now().UTC())
```
and somewhere else :
```
func CreateEventMap(apiName string /* and other generic parameters... */) map[string]string {
}
func AddEventValues(eventNfo map[string]string, values string...) {
    // can automatically ignore key/value when value is nil
}
func AddEventTime(eventNfo map[string]string, fieldName string, time time.Time) {
}
```
To illustrate what is worrying me, the initial DeleteUser function is only 2 lines long to do what it is aimed to but is more than 20 lines with code not related to user deletion.